subprojects/core/src/main/java/org/gradle/plugin/management/PluginResolutionSpec.java

```diff
@@ -0,0 +1,15 @@
+package org.gradle.plugin.management;
+
+import org.gradle.api.Incubating;
+
+/**
+ * This needs a better name.
+ *
+ * Allows user to specify details about the plugin that needs
+ * to be loaded.
+ */
+@Incubating
+public interface PluginResolutionSpec {
+
+    void usePluginName(String name);
```

Comment:
Can we leave this feature for a follow-up PR? I'd like to keep this one as small as we can.